File: app/src/main/java/com/ed/redditapp/ui/search/SearchModel.java
package com.ed.redditapp.ui.search;

import com.ed.redditapp.lib.api.RedditApi;
import com.ed.redditapp.lib.api.SearchItemSubreddit;

import lombok.Getter;

public class SearchModel {
    private final SearchAsyncTask searchAsyncTask;

    @Getter
    private SearchItemSubreddit[] subreddits;

    public SearchModel(RedditApi redditApi) {
        this.searchAsyncTask = new SearchAsyncTask(redditApi);
    }

    public void searchSubreddits(String searchText, SearchAsyncTask.Callback callback) {
        searchAsyncTask.run(searchText, subreddits -> {
            this.subreddits = subreddits;
            callback.callback(subreddits);
        });
    }
}

File: app/src/main/java/com/ed/redditapp/ui/search/SearchModel.kt
package com.ed.redditapp.ui.search

import com.ed.redditapp.lib.api.RedditApi
import com.ed.redditapp.lib.api.SearchItemSubreddit
import lombok.Getter

class SearchModel(redditApi: RedditApi?) {
    private val searchAsyncTask = SearchAsyncTask(redditApi!!)

    var subreddits: Array<SearchItemSubreddit> = emptyArray()
        private set

    fun searchSubreddits(searchText: String, callback: SearchAsyncCallback) {
        searchAsyncTask.run(searchText) { subreddits ->
            this.subreddits = subreddits
            callback(subreddits)
        }
    }
}